executor/builder.go

```diff
@@ -856,29 +862,16 @@ func (b *executorBuilder) buildTableReader(v *plan.PhysicalTableReader) Executor
 		ranges:    ts.Ranges,
 	}
 
-	for _, col := range v.Schema().Columns {
-		dagReq.OutputOffsets = append(dagReq.OutputOffsets, uint32(col.Index))
+	for i := range v.Schema().Columns {
+		dagReq.OutputOffsets = append(dagReq.OutputOffsets, uint32(i))
```

Comment:
@zimulala For Index Reader, we have the right index because we resolve index again when constructing index reader. It is a little tricky. The right way is to resolve indices **after** physical plan building. This way will change a lot and I will refactor it in the future.